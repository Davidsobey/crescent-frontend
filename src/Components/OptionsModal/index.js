import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Typography } from 'material-ui';
import Radio, { RadioGroup } from 'material-ui/Radio';
import { FormLabel, FormControl, FormControlLabel } from 'material-ui/Form';

import ModalPaper from './styles';
import Button from '../Button';

class OptionsModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: '0',
    };
  }

  handleChange = event => {
    this.setState({ value: event.target.value });
  };

  render() {
    const style = {
      justifyContent: 'center',
      alignItems: 'center',
    };
    return (
      <Modal
        aria-labelledby="simple-modal-title"
        aria-describedby="simple-modal-description"
        open={this.props.open}
        style={style}
      >
        <ModalPaper>
          <Typography variant="title" id="modal-title">
            {this.props.title}
          </Typography>
          <br />
          <FormControl component="fieldset">
            <FormLabel component="legend">
              Select an option
            </FormLabel>
            <RadioGroup
              aria-label="select an option"
              name="option"
              value={this.state.value}
              onChange={this.handleChange}
            >
              {this.props.options.map((option, index) => (
                <FormControlLabel
                  value={`${index}`}
                  control={<Radio />}
                  label={option.label}
                  key={index}
                />
              ))}
            </RadioGroup>
          </FormControl>
          <br />
          <div className="justify-content flex-container">
            <Button
              variant="raised"
              color="primary"
              onClick={() => this.props.onClick(this.state.value)}
            >
              Continue
            </Button>
          </div>
        </ModalPaper>
      </Modal>
    );
  }
}

OptionsModal.propTypes = {
  onClick: PropTypes.func,
  title: PropTypes.string,
  open: PropTypes.bool,
  options: PropTypes.array,
};

export default OptionsModal;
