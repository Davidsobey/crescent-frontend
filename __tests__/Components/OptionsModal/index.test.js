import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import OptionsModal from '../../../src/Components/OptionsModal/index.js';


const mockStore = configureStore();
const initialState = {};
const store = mockStore(initialState);

const options = [
  // {label: 'Create another assessment'},
  {label: 'Create another module', url: '/module/create'},
  {label: 'Create questions for this assessment', url: '/question/create'},
];

describe('<OptionsModal />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<OptionsModal store={store} options={options} open={false} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});