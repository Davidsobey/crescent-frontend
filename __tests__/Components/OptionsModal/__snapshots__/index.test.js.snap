// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionsModal /> render() renders the component 1`] = `
<Modal
  BackdropComponent={[Function]}
  aria-describedby="simple-modal-description"
  aria-labelledby="simple-modal-title"
  classes={
    Object {
      "hidden": "MuiModal-hidden-2",
      "root": "MuiModal-root-1",
    }
  }
  disableAutoFocus={false}
  disableBackdropClick={false}
  disableEnforceFocus={false}
  disableEscapeKeyDown={false}
  disableRestoreFocus={false}
  hideBackdrop={false}
  keepMounted={false}
  manager={
    ModalManager {
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "modals": Array [],
    }
  }
  open={false}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <styled.div>
    <WithStyles(Typography)
      id="modal-title"
      variant="title"
    />
    <br />
    <WithStyles(FormControl)
      component="fieldset"
    >
      <WithStyles(FormLabel)
        component="legend"
      >
        Select an option
      </WithStyles(FormLabel)>
      <RadioGroup
        aria-label="select an option"
        name="option"
        onChange={[Function]}
        value="0"
      >
        <WithStyles(FormControlLabel)
          control={<WithStyles(Radio) />}
          key="0"
          label="Create another module"
          value="0"
        />
        <WithStyles(FormControlLabel)
          control={<WithStyles(Radio) />}
          key="1"
          label="Create questions for this assignment"
          value="1"
        />
      </RadioGroup>
    </WithStyles(FormControl)>
    <br />
    <div
      className="justify-content flex-container"
    >
      <WithTheme(ButtonWithTheme)
        color="primary"
        onClick={[Function]}
        variant="raised"
      >
        Continue
      </WithTheme(ButtonWithTheme)>
    </div>
  </styled.div>
</Modal>
`;
